src/WebJobs.Script/Config/INameResolverExtensions.cs

```diff
@@ -0,0 +1,37 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using Microsoft.Azure.WebJobs.Host;
+using System;
+using System.Reflection;
+
+namespace Microsoft.Azure.WebJobs.Script.Description
+{
+    // Attribute to mark a string property has embedded %key% and supports INameResolver resolution. 
+    [AttributeUsage(AttributeTargets.Property)]
+    public class AllowNameResolutionAttribute : Attribute
```

Comment:
I don't see this attribute used anywhere? Recommend removing this and the below extension method. If you're considering introducing this pattern, it might belong in the core SDK if I understand the intent correctly. Currently we declaratively resolve all attribute properties that support name resolution, and this appears to be a way to declaratively do that. If so, I recommend adding a work item for the core SDK for this - when it gets done, we'd move all our existing attributes to use this.